### PR TITLE
Project Restructuring

### PR DESCRIPTION
**Pull request by stefan-o-dimitrov**
- Moved pieces and related classes to jchess.pieces
- Moved UI-related files, excluding JChessView and GUI, to jchess.UI
- Moved classes Chessboard and Square to jchess.UI.board

### CODE REVIEW

File: jchess/GUI.java [hunk 1]
```diff
@@ -26,9 +26,11 @@
 import java.io.InputStreamReader;
 import javax.swing.*;
 import javax.swing.JPanel;
```
Comment (frozencure) [inline jchess/GUI.java]: Shouldn't this class also belong to the `UI` package?

File: jchess/JChessView.java [hunk 2]
```diff
@@ -31,14 +40,15 @@
...
 import java.applet.*;
 import java.io.IOException;
+import java.util.logging.Level;
 
 
 
 /**
  * The application's main frame.
  */
 public class JChessView extends FrameView implements ActionListener, ComponentListener
```
Comment (frozencure) [inline jchess/JChessView.java]: Doesn't this class also belong to the `UI` package?

File: jchess/Log.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import java.io.IOException;
+import java.util.logging.FileHandler;
+import java.util.logging.Level;
+import java.util.logging.SimpleFormatter;
+
+public class Log {
+	private static final Logger LOGGER = Logger.getLogger(Log.class.getName());
+	private static FileHandler fh;
+	
+	public static void init() {
```
Comment (frozencure) [inline jchess/Log.java]: What is the `init()` method supposed to do? It is not called anywhere. (btw i like that you used a singleton)

File: jchess/UI/Chat.java [hunk 4]
```diff
@@ -27,6 +27,9 @@
 import javax.swing.JPanel;
 import javax.swing.JTextArea;
 import javax.swing.JTextField;
+
+import jchess.Client;
```
Comment (frozencure) [inline jchess/UI/Chat.java]: Please remove unnecessary lines

File: jchess/UI/Clock.java [hunk 5]
```diff
@@ -18,7 +18,9 @@
  * Mateusz Sławomir Lach ( matlak, msl )
  * Damian Marciniak
  */
-package jchess;
+package jchess.UI;
+
```
Comment (frozencure) [inline jchess/UI/Clock.java]: Remove (auto-format)

File: jchess/UI/GameClock.java [hunk 6]
```diff
@@ -18,13 +18,20 @@
  * Mateusz Sławomir Lach ( matlak, msl )
  * Damian Marciniak
  */
-package jchess;
+package jchess.UI;
 
 import java.awt.*;
 import java.awt.image.*;
+import java.util.logging.Level;
+
```
Comment (frozencure) [inline jchess/UI/GameClock.java]: Auto-format

File: jchess/UI/GameClock.java [hunk 7]
```diff
@@ -18,13 +18,20 @@
...
 import javax.swing.JOptionPane;
 import javax.swing.JPanel;
 
+import jchess.Game;
+import jchess.Log;
+import jchess.Player;
+import jchess.Settings;
+
```
Comment (frozencure) [inline jchess/UI/GameClock.java]: Remove (auto-format)

File: jchess/UI/JChessAboutBox.java [hunk 8]
```diff
@@ -13,10 +13,12 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package jchess;
+package jchess.UI;
 
 import org.jdesktop.application.Action;
 
+import jchess.JChessApp;
+
```
Comment (frozencure) [inline jchess/UI/JChessAboutBox.java]: Remove (auto-format)

File: jchess/UI/JChessTabbedPane.java [hunk 9]
```diff
@@ -18,9 +18,14 @@
  * Mateusz Sławomir Lach ( matlak, msl )
  * Damian Marciniak
  */
-package jchess;
+package jchess.UI;
 
 import javax.swing.*;
+
```
Comment (frozencure) [inline jchess/UI/JChessTabbedPane.java]: Remove (auto-format)

File: jchess/UI/NewGameWindow.java [hunk 10]
```diff
@@ -18,9 +18,12 @@
  *
  * Created on 2009-10-20, 15:11:49
  */
-package jchess;
+package jchess.UI;
 
 import javax.swing.*;
+
```
Comment (frozencure) [inline jchess/UI/NewGameWindow.java]: Remove (auto-format)

File: jchess/UI/PawnPromotionWindow.java [hunk 11]
```diff
@@ -18,10 +18,13 @@
  * Mateusz Sławomir Lach ( matlak, msl )
  * Damian Marciniak
  */
-package jchess;
+package jchess.UI;
 
 import java.awt.*;
 import javax.swing.*;
+
```
Comment (frozencure) [inline jchess/UI/PawnPromotionWindow.java]: Remove (auto-format)

File: jchess/UI/board/Chessboard.java [hunk 12]
```diff
@@ -27,8 +27,27 @@
 import java.awt.image.BufferedImage;
 import java.util.ArrayList;
 import java.util.Iterator;
+import java.util.logging.Level;
+
 import javax.swing.JPanel;
-import jchess.Moves.castling;
+
```
Comment (frozencure) [inline jchess/UI/board/Chessboard.java]: Remove (auto-format)

File: jchess/Log.java (hunk 3 above)
Comment (stefan-o-dimitrov) [reply jchess/Log.java]: It's supposed to be called at startup, so the logger logs in a file and not the console, but I forgot to add a call to init() 😅 Adding it in `JChessApp`'s `startup` method. :)

File: jchess/UI/Chat.java (hunk 4 above)
Comment (stefan-o-dimitrov) [reply jchess/UI/Chat.java]: The ones around the `import` statement?

File: jchess/UI/Clock.java (hunk 5 above)
Comment (stefan-o-dimitrov) [reply jchess/UI/Clock.java]: No new line after `package` statement? Goes for the ones below as well.

File: jchess/UI/Chat.java (hunk 4 above)
Comment (frozencure) [reply jchess/UI/Chat.java]: Yes, this is the same for every comment below. Just perform an auto-format in eclipse and all the empty lines will disappear. (`Ctrl` + `Shift` + `F`)

File: jchess/GUI.java (hunk 1 above)
Comment (stefan-o-dimitrov) [reply jchess/GUI.java]: Yeah, this one generates weird Exceptions when it's in `UI`, I think resource paths need to be changed or something, not sure.

Leaving it here for now.

File: jchess/JChessView.java (hunk 2 above)
Comment (stefan-o-dimitrov) [reply jchess/JChessView.java]: Same as `GUI`.

**Pull request merged**